etree.go

```diff
@@ -1033,28 +1031,28 @@ func (e *Element) setIndex(index int) {
 }
 
 // writeTo serializes the element to the writer w.
-func (e *Element) writeTo(w *bufio.Writer, s *WriteSettings) {
-	w.WriteByte('<')
+func (e *Element) writeTo(w io.StringWriter, s *WriteSettings) {
+	w.WriteString("<")
```

Comment:
Why the switch to `io.StringWriter` from `bufio.Writer`?  The use of `bufio.Writer` was never exposed in the public API.  Also, there are subtle performance consequences to this change. For instance, `bufio.WriteByte` is faster than `bufio.WriteString` for single-character strings.